test/runtests.jl

```diff
@@ -1,36 +1,14 @@
 using Test
 
-@testset "MCMCChains" begin
```

Comment:
This is very difficult to read -- can we maintain the original formatting?

Reply:
The problem with having one MCMCChains testset is that there is no information (and update while running) of the nested testsets. Moreover, all tests will be run even if already the first testset errors. Additionally, even if everything runs reasonably fast, we might have to introduce print statements to just to prevent Travis from cancelling the job.

Reply:
We can keep the separate test sets, but they should be nested within an `MCMCChains` set, and should all be on separate lines so they are easier to read. Something like

```julia
@testset "MCMCChains" begin
    @testset "Diagnostics" begin 
        include("diagnostic_tests.jl") 
    end

    . . .
end
```

We should also probably not have `@time` calls in there, but I presume that's temporary for testing purpose.

Reply:
> they should be nested within an MCMCChains set

Even if that leads to no information about the individual testsets, possibly requires print statements to wake up Travis, and means that tests are not stopped after a failing testset?

> We should also probably not have @time calls in there, but I presume that's temporary for testing purpose.

If everything is nested inside one single testset, the `@time` calls are not useful at all. In the other case, they would make it easier to spot performance regressions and which tests and function calls actually lead to many allocations and bad performance.

Reply:
> Even if that leads to no information about the individual testsets, possibly requires print statements to wake up Travis, and means that tests are not stopped after a failing testset?

Yes. Tests should proceed if one test set fails -- we want to know everywhere something fails, not the first place it fails. I'm alright with `println` statements if you want to indicate what's being tested -- Turing's tests have something similar.

I'm fine with leaving in the `@time` statements.